File: Beginner/Java/String/FirstUniqChar.java
class Solution {
    public int firstUniqChar(String s) {
        int[] c = new int[26];
        int minPos = s.length();
        for(int i = 0; i < s.length(); i ++)
            c[s.charAt(i) - 'a'] ++;
        for(int i = 0; i < 26; i ++)
            if(c[i] == 1)
                minPos = Math.min(minPos, s.indexOf(i + 'a'));
        if (minPos == s.length())
            return -1;
        return minPos;
    }
}

File: Beginner/C/String/FirstUniqChar.c
int firstUniqChar(char* s) {
    int count[26];
    char* p = s;
    for(int i = 0; i < 26; i ++)
        count[i] = 0;
    while(*p != 0 && p) {
        count[((int)*p) - 97] ++;
        p ++;
    }
    int pos = 0;
    while(*s != 0 && s) {
        if(count[((int)*s) - 97] == 1)
            return pos;
        pos ++;
        s ++;
    }
    return -1;
}
